### PR TITLE
Book on Testing JS applications

### PR DESCRIPTION
**Pull request by SampurnaSen**
## What does this PR do?
Add resource(s)

### Description
Testing JavaScript Applications is a book to learn effective tests through examples. It covers tools that teach how to think about testing from a business perspective.
### Why is this valuable (or not)?
This book will help you write high-quality software in less time, with more confidence.
### How do we know it's really free?
Available in GitHub for free.
### For book lists, is it a book? For course lists, is it a course? etc.
It is a book.
## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -1127,6 +1127,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [Single page apps in depth](http://singlepageappbook.com) - Mixu (HTML)
 * [Speaking JavaScript](http://speakingjs.com/es5/) - Dr. Axel Rauschmayer
 * [Standard ECMA-262 ECMAScript 2016 Language Specification](https://www.ecma-international.org/publications/standards/Ecma-262.htm) - Ecma International (HTML,PDF)
+* [Testing JavaScript Applications](https://github.com/lucasfcosta/testing-javascript-applications) - Lucas da Costa
```
Comment (davorpa) [inline books/free-programming-books-langs.md:1130]:
```suggestion
* [Testing JavaScript Applications](https://github.com/lucasfcosta/testing-javascript-applications) - Lucas Fernandes da Costa
```

Where is the book? Is not free at Manning... https://www.manning.com/books/testing-javascript-applications

**Comment by SampurnaSen** [comment]
In case you approve my PR, could you please tag my pr with the hacktoberfest-accepted label. Much appreciated!

File: books/free-programming-books-langs.md (hunk 1 above)
Comment (SampurnaSen) [reply books/free-programming-books-langs.md]: No, it's free on GitHub.

**Comment by eshellman** [comment]
github content is not the book, which is not free

**Pull request closed without merging**